bookkeeper-server/src/main/java/org/apache/bookkeeper/client/DefaultEnsemblePlacementPolicy.java

```diff
@@ -51,18 +66,43 @@
             return newBookies;
         }
         List<BookieSocketAddress> allBookies;
-        synchronized (this) {
+        rwLock.readLock().lock();
+        try {
             allBookies = new ArrayList<BookieSocketAddress>(knownBookies);
+        } finally {
+            rwLock.readLock().unlock();
```

Comment:
Do we need to consider the race that  'knownBookies' is changed between this line and line 79?